File: app/src/main/java/com/irishinterest/irishinterest/fragments/authors/AuthorsProvider.java
package com.irishinterest.irishinterest.fragments.authors;

import com.irishinterest.irishinterest.model.authors.AuthorValues;
import com.irishinterest.irishinterest.model.provider.Provider;
import com.irishinterest.irishinterest.network.api.irishInterest.NotificationOnChange;
import com.irishinterest.irishinterest.network.api.irishInterest.observer.GuiMessage;
import com.irishinterest.irishinterest.network.api.irishInterest.observer.GuiToModel;
import com.irishinterest.irishinterest.network.api.irishInterest.observer.Module;

import java.util.ArrayList;
import java.util.Observable;
import java.util.Observer;

public class AuthorsProvider extends Provider<AuthorValues> implements Observer {

    private AuthorValues authors;

    public AuthorsProvider(GuiToModel guiToModel, NotificationOnChange<AuthorValues> notificationOnChange) {
        super(guiToModel, notificationOnChange);
        this.notificationOnChange.registerProvider(this);
        this.guiToModel.registerObserver(Module.AUTHORS, this);
    }

    @Override
    protected ArrayList<AuthorValues> getData() {
        return null;
    }

    @Override
    protected void processMessage(GuiMessage<AuthorValues> guiMessage) {

    }

    @Override
    public void update(Observable o, Object arg) {

    }
}

File: app/src/main/java/com/irishinterest/irishinterest/fragments/authors/AuthorsProvider.kt
package com.irishinterest.irishinterest.fragments.authors

import com.irishinterest.irishinterest.network.api.irishInterest.observer.GuiToModel
import com.irishinterest.irishinterest.network.api.irishInterest.NotificationOnChange
import com.irishinterest.irishinterest.model.authors.AuthorValues
import com.irishinterest.irishinterest.model.provider.Provider
import com.irishinterest.irishinterest.network.api.irishInterest.observer.GuiMessage
import com.irishinterest.irishinterest.network.api.irishInterest.observer.Module
import java.util.*
import kotlin.collections.ArrayList

class AuthorsProvider(
    guiToModel: GuiToModel,
    notificationOnChange: NotificationOnChange<AuthorValues>,
    override val data: ArrayList<AuthorValues> = ArrayList<AuthorValues>()
) : Provider<AuthorValues>(
    guiToModel, notificationOnChange
), Observer {
    protected override fun processMessage(guiMessage: GuiMessage<AuthorValues>?) {}
    override fun update(o: Observable, arg: Any) {}

    init {
        this.notificationOnChange.registerProvider(this)
        this.guiToModel.registerObserver(Module.AUTHORS, this)
    }
}